### PR TITLE
Use Swift syntax in reference documentation

### LINKED ISSUE #9588: Update all docs from Objective C APIs and types to Swift
Assigning to @morganchen12 since he said he was working on this.

### PR DESCRIPTION
**Pull request by morganchen12**
Fixes #9588. #no-changelog

### CODE REVIEW

File: FirebaseAuth/Sources/Public/FirebaseAuth/FIRGitHubAuthProvider.h [hunk 1]
```diff
@@ -37,7 +37,7 @@ NS_SWIFT_NAME(GitHubAuthProvider)
 @interface FIRGitHubAuthProvider : NSObject
 
 /** @fn credentialWithToken:
-    @brief Creates an `FIRAuthCredential` for a GitHub sign in.
+    @brief Creates an `AuthCredential` for a GitHub sign in.
 
     @param token The GitHub OAuth access token.
     @return A FIRAuthCredential containing the GitHub credential.
```
Comment (ryanwilson) [inline FirebaseAuth/Sources/Public/FirebaseAuth/FIRGitHubAuthProvider.h:43]: Nit: missed this one

File: FirebaseDatabase/Sources/Public/FirebaseDatabase/FIRTransactionResult.h [hunk 2]
```diff
@@ -20,28 +20,28 @@
...
 /**
- * Used for runTransactionBlock:. Indicates that the current transaction should
- * no longer proceed.
+ * Used for `runTransactionBlock(_:)`. Indicates that the current transaction
+ * should no longer proceed.
  *
- * @return An FIRTransactionResult instance that can be used as a return value
- * from the block given to runTransactionBlock:
+ * @return A `TransactionResult` instance that can be used as a return value
+ * from the block given to runTransactionBlock(_:)
```
Comment (ryanwilson) [inline FirebaseDatabase/Sources/Public/FirebaseDatabase/FIRTransactionResult.h:44]: Nit: backticks around the function call

File: FirebaseInAppMessaging/Sources/Public/FirebaseInAppMessaging/FIRInAppMessaging.h [hunk 3]
```diff
@@ -24,20 +24,20 @@ NS_ASSUME_NONNULL_BEGIN
 /**
  * The root object for in-app messaging iOS SDK.
  *
- * Note: Firebase InApp Messaging depends on using a Firebase Instance ID & token pair to be able
- * to retrieve FIAM messages defined for the current app instance. By default, Firebase in-app
- * messaging SDK would obtain the ID & token pair on app/SDK startup. As a result of using
- * ID & token pair, some device client data (linked to the instance ID) would be collected and sent
- * over to Firebase backend periodically.
+ * Note: Firebase In-App Messaging depends on using a Firebase Installation ID and token pair to be
+ *able to retrieve messages defined for the current app instance. By default, the Firebase In-App
```
Comment (ryanwilson) [inline FirebaseInAppMessaging/Sources/Public/FirebaseInAppMessaging/FIRInAppMessaging.h:28]: Nit: space before `able`

**Review by ryanwilson — COMMENTED** [review]
Couple nits but otherwise LGTM

File: FirebaseInAppMessaging/Sources/Public/FirebaseInAppMessaging/FIRInAppMessaging.h (hunk 3 above)
Comment (morganchen12) [reply FirebaseInAppMessaging/Sources/Public/FirebaseInAppMessaging/FIRInAppMessaging.h]: Done.

File: FirebaseDatabase/Sources/Public/FirebaseDatabase/FIRTransactionResult.h (hunk 2 above)
Comment (morganchen12) [reply FirebaseDatabase/Sources/Public/FirebaseDatabase/FIRTransactionResult.h]: Done.

File: FirebaseAuth/Sources/Public/FirebaseAuth/FIRGitHubAuthProvider.h (hunk 1 above)
Comment (morganchen12) [reply FirebaseAuth/Sources/Public/FirebaseAuth/FIRGitHubAuthProvider.h]: Done.

**Comment by paulb777** [comment]
cc: @eldhosembabu @rosalyntan @IanWyszynski @tonyjhuang @karenyz @visumickey @elenadoty

File: FirebaseStorage/Sources/Public/FirebaseStorage/FIRStorageReference.h [hunk 4]
```diff
@@ -35,25 +35,25 @@ NS_ASSUME_NONNULL_BEGIN
 @interface FIRIMPLStorageReference : NSObject
```
Comment (paulb777) [inline FirebaseStorage/Sources/Public/FirebaseStorage/FIRStorageReference.h:35]: This is no longer a public header and will cause a messy conflict with #9598. Please revert

**Review by paulb777 — APPROVED** [review]
LGTM after reverting the Storage change.

File: FirebaseStorage/Sources/Public/FirebaseStorage/FIRStorageReference.h (hunk 4 above)
Comment (morganchen12) [reply FirebaseStorage/Sources/Public/FirebaseStorage/FIRStorageReference.h]: Done.

**Comment by morganchen12** [comment]
Ignoring non-style test failures since this PR changes comments only.

**Pull request merged**